pkgs/development/libraries/arrayfire/default.nix

```diff
@@ -1,91 +1,183 @@
-{ lib
-, stdenv
-, fetchFromGitHub
+{ blas
+, boost
 , cmake
-, pkg-config
-, opencl-clhpp
-, ocl-icd
+, config
+, cudaPackages
+, fetchFromGitHub
 , fftw
 , fftwFloat
-, blas
+, fmt_9
+, forge
+, freeimage
+, git
+, gtest
 , lapack
-, boost
-, mesa
-, libGLU
+, lib
 , libGL
-, freeimage
+, mesa
+, ocl-icd
+, opencl-clhpp
+, pkg-config
 , python3
-, clfft
-, clblas
-, doxygen
-, buildDocs ? false
-, config
-, cudaSupport ? config.cudaSupport
-, cudatoolkit
-, darwin
+, span-lite
+, stdenv
+, doCheck ? false
+, withCPU ? true
+, withCuda ? config.cudaSupport
+, withOpenCL ? stdenv.isLinux
+, nvidiaComputeDrivers ? null
 }:
 
+assert blas.isILP64 == false;
+
 stdenv.mkDerivation rec {
   pname = "arrayfire";
-  version = "3.7.3";
+  version = "3.9.0-pre";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
-    rev = "v${version}";
-    sha256 = "0gcbg6b6gs38xhks5pp0vkcqs89zl7rh9982jqlzsd0h724qddw0";
-    fetchSubmodules = true;
+    rev = "d2a66367d859cdb554f2374e29d39c88d5fff978";
+    hash = "sha256-9r1w0U9MvhduHwBpEWpqkrQPawd94EY3FAqSJghi09I=";
+  };
+
+  assets = fetchFromGitHub {
+    owner = pname;
+    repo = "assets";
+    rev = "cd08d749611b324012555ad6f23fd76c5465bd6c";
+    sha256 = "sha256-v4uhqPz1P1g1430FTmMp22xJS50bb5hZTeEX49GgMWg=";
+  };
+  clblast = fetchFromGitHub {
+    owner = "cnugteren";
+    repo = "CLBlast";
+    rev = "4500a03440e2cc54998c0edab366babf5e504d67";
+    sha256 = "sha256-I25ylQp6kHZx6Q7Ph5r3abWlQ6yeIHIDdS1eGCyArZ0=";
+  };
+  clfft = fetchFromGitHub {
+    owner = pname;
+    repo = "clfft";
+    rev = "760096b37dcc4f18ccd1aac53f3501a83b83449c";
+    sha256 = "sha256-vJo1YfC2AJIbbRj/zTfcOUmi0Oj9v64NfA9MfK8ecoY=";
+  };
+  glad = fetchFromGitHub {
+    owner = pname;
+    repo = "glad";
+    rev = "ef8c5508e72456b714820c98e034d9a55b970650";
+    sha256 = "sha256-u9Vec7XLhE3xW9vzM7uuf+b18wZsh/VMtGbB6nMVlno=";
+  };
+  threads = fetchFromGitHub {
+    owner = pname;
+    repo = "threads";
+    rev = "4d4a4f0384d1ac2f25b2c4fc1d57b9e25f4d6818";
+    sha256 = "sha256-qqsT9woJDtQvzuV323OYXm68pExygYs/+zZNmg2sN34=";
+  };
+  test-data = fetchFromGitHub {
+    owner = pname;
+    repo = "arrayfire-data";
+    rev = "a5f533d7b864a4d8f0dd7c9aaad5ff06018c4867";
+    sha256 = "sha256-AWzhsrDXyZrQN2bd0Ng/XlE8v02x7QWTiFTyaAuRXSw=";
+  };
+  cub = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "cub";
+    rev = "1.10.0";
+    sha256 = "sha256-JyyNaTrtoSGiMP7tVUu9lFL07lyfJzRTVtx8yGy6/BI=";
+  };
```

Comment:
This, iirc, is provided by `cudaPackages.cccl` (or something like that; also used by the nvidia-thrust derivation). Is it necessary to `cp -rf` the whole tree, or is it possible to discover it properly using e.g. CMake?

Reply:
@SomeoneSerge good catch, thanks! Fixed. I've tested the change locally on a CUDA-capable GPU, and the tests still pass.

Reply:
I don't see the "mark resolved" button, I guess you have it